syntax = "proto3";

package stq;

import "types.proto";

service Queue {
  rpc CreateQueue(QueueReq) returns (Empty);
  rpc RenameQueue(RenameQueueReq) returns (Empty);
  rpc DeleteQueue(QueueReq) returns (Empty);
  rpc ListQueue(ListQueueReq) returns (ListQueueRes);
}

message QueueReq {
  string name = 1;
}

message RenameQueueReq {
  string oldname = 1;
  string newName = 2;
}

message ListQueueReq {
  int32 startIndex = 1;
  int32 limit = 2;
}

message ListQueueRes {
  StateCode stateCode = 1;
  repeated string list = 2;
}
